core/domain/exp_jobs_one_off.py

```diff
@@ -522,10 +521,16 @@ def map(item):
                         err_dict[tag.name] = [parent]
 
         for key in err_dict:
-            err_dict[key] = list(set(err_dict[key]))
-
-        yield('Errors', err_dict)
+            yield(key, list(set(err_dict[key])))
 
     @staticmethod
     def reduce(key, values):
-        yield (key, values)
+        final_values = []
```

Comment:
Um, I don't think this is the way to do it. Look at other reduce() methods in MR jobs; use ast.literal_eval() to parse.

Also, in general, try to avoid redefining variables (like you've done in lines 530 and 531). Otherwise it becomes hard to tell what "value" represents as its meaning keeps changing.

Reply:
Fixed this!

Reply:
Also, thanks a lot for all the suggestions, they had really made the code and the output much cleaner. `ast.literal_eval()` has made the output better :smile: .

Reply:
You're welcome! Glad to help.